File: ts/reverseWords.ts
function reverseWords(s: string): string {
  return s
    .split(' ')
    .filter(it => !!it.length)
    .reverse()
    .join(' ');
}

export default () => {
  const s = 'a good   example';
  const result = reverseWords(s);
  console.log(result);
};

File: rs/src/reverse_words.rs
fn reverse_words(s: String) -> String {
    return s.split_whitespace().rev().collect::<Vec<&str>>().join(" ");
}

pub fn main() {
    // let s: String = "a good example".to_string();
    let s: String = "the sky is blue".to_string();
    let result = reverse_words(s);

    println!("{result}");
}
